operations/commit_queue.go

```diff
@@ -245,6 +285,62 @@ func (p *mergeParams) uploadMergePatch(conf *ClientSettings, ac *legacyClient) e
 	return nil
 }
 
+type moduleParams struct {
+	patchID     string
+	module      string
+	ref         string
+	large       bool
+	skipConfirm bool
+}
+
+func (p *moduleParams) addModule(ac *legacyClient, rc *legacyClient) error {
```

Comment:
I might break this apart. right now the confirms would make it imposslbe to call this code as a test.

Reply:
done. I would argue, however, the confirms are surmountable by passing `p.skipConfirm == true`

Reply:
I find shelling out to git to be an issue, though.